src/cocotb/triggers.py

```diff
@@ -701,137 +735,125 @@ async def coro_inner():
         result = await Join(task)
         assert result == "Hello world"
 
-    If the coroutine threw an exception, the :keyword:`await` will re-raise it.
-
+    .. note::
+        There is typically no reason to directly use this trigger.
```

Comment:
```suggestion
        Typically there is no reason to directly use this trigger.
```
(Use wording of `retval` attribute below.)